src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigStorageLocation`1.cs

```diff
@@ -54,23 +54,30 @@ public bool TryGetOption(object underlyingOption, IReadOnlyDictionary<string, ob
         {
             if (allRawConventions.TryGetValue(KeyName, out object value))
             {
-                var optionalValue = _parseValue(value.ToString(), type);
-                if (optionalValue.HasValue)
-                {
-                    result = optionalValue.Value;
-                }
-                else
-                {
-                    result = null;
-                }
-
-                return result != null;
+                var ret = TryGetOption(value.ToString(), type, out var typedResult);
```

Comment:
does ToString() here allocate? or are we assuming most of time it is string so ToString won't allocate?

Reply:
➡️ This is not the method we are using on the hot path. The `ToString()` call existed prior to my change.